frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { compose } from 'redux';
+
+import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import DetailItem from '@console/shared/src/components/dashboard/details-card/DetailItem';
+import DetailsBody from '@console/shared/src/components/dashboard/details-card/DetailsBody';
+import { usePrometheusQueries } from '@console/shared/src/components/dashboard/utilization-card/prometheus-hook';
+import {
+  humanizePercentage,
+  humanizeBinaryBytes,
+  humanizeNumber,
+} from '@console/internal/components/utils';
+
+import { BlockPoolDashboardContext } from './block-pool-dashboard-context';
+import { getPoolQuery, StorageDashboardQuery } from '../../../queries/ceph-queries';
+
+const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
+
+export const CompressionDetailsCard: React.FC = () => {
+  const { t } = useTranslation();
+  const { obj } = React.useContext(BlockPoolDashboardContext);
+  const compressionMode = obj.spec?.compressionMode;
+  const compressionEnabled = compressionMode !== 'none';
+  const { name } = obj.metadata;
+
+  // Compression Metrics
+  const queries = React.useMemo(
+    () => [
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_SAVINGS),
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_ELIGIBILITY),
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_RATIO),
+    ],
+    [name],
+  );
+
+  const [values, loading] = usePrometheusQueries(queries, parser as any);
+  const poolCompressionSavings = (values?.[0] ?? 0) as number;
+  const poolCompressionEligibility = (values?.[1] ?? 0) as number;
+  const poolCompressionRatio = (values?.[2] ?? 0) as number;
+
+  return (
+    <DashboardCard data-test-id="compression-details-card">
+      <DashboardCardHeader>
+        <DashboardCardTitle>{t('ceph-storage-plugin~Compression')}</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression status')}>
+            {!compressionEnabled ? 'Disabled' : 'Enabled'}
+          </DetailItem>
+          {compressionEnabled && (
+            <>
+              <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage efficiency')}>
+                {''}
```

Comment:
why empty ?
can you share screenshot as well ?

Reply:
The compression card itself

![Screenshot from 2021-07-21 13-20-35](https://user-images.githubusercontent.com/895234/126457600-a18b62d5-6520-4fac-bfba-f43ed29836d8.png)

Compression card in the dashboard page

![Screenshot from 2021-07-21 13-19-40](https://user-images.githubusercontent.com/895234/126457644-6bda560d-f27f-40a4-ad86-9de60afb2ad2.png)

Reply:
@GowthamShanmugam , @afreen23 , please take a look at the screen shot

Reply:
@afreen23 , for your above comment (https://github.com/openshift/console/pull/9507#discussion_r673313723), I have updated the code a bit differently.

Removed the default ZERO value, as we won't enter into the default state if the query is still being loaded (or we hit any query error). Please take a look.